konrad/core.py

```diff
@@ -201,6 +206,13 @@ def run(self):
                 surface=self.atmosphere.surface,
                 )
 
+            # Update the humidity profile.
+            self.atmosphere['H2O'][0, :] = self.humidity.get(
+                    self.atmosphere,
+                    p_tropo=self.atmosphere.get_convective_top(
```

Comment:
We could consider to pass the `heatingrates` to the `get()`-method and calculate the convective top inside. This way, it would be possible to only calculate `p_tropo` if it is required (this would need some refactor of the existing humidity classes, but I want to do that anyhow).

For now it is just a matter of taste, so you can decide 😉